File: app/src/main/java/com/bitchat/android/BitchatApplication.java
package com.bitchat.android;

import android.app.Application;
import com.bitchat.android.favorites.FavoritesPersistenceService;
import com.bitchat.android.nostr.NostrIdentityBridge;
import com.bitchat.android.nostr.RelayDirectory;

/**
 * Classe principale de l'application pour bitchat Android.
 */
public class BitchatApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // Initialiser le répertoire des relais (charge assets/nostr_relays.csv)
        RelayDirectory.initialize(this);

        // Initialiser la persistance des favoris tôt
        try {
            FavoritesPersistenceService.initialize(this);
        } catch (Exception e) {
            // Ignorer l'erreur
        }

        // Préchauffer l'identité Nostr pour s'assurer que npub est disponible
        try {
            NostrIdentityBridge.getCurrentNostrIdentity(this);
        } catch (Exception e) {
            // Ignorer l'erreur
        }

        // La logique pour initialiser le thème et le gestionnaire de préférences de débogage irait ici.
    }
}

File: app/src/main/java/com/bitchat/android/BitchatApplication.kt
package com.bitchat.android

import android.app.Application
import com.bitchat.android.di.initKoin
import com.bitchat.android.ui.theme.ThemePreferenceManager
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger

/**
 * Main application class for bitchat Android
 */
class BitchatApplication : Application() {
    
    override fun onCreate() {
        super.onCreate()

        initKoin{
            androidContext(this@BitchatApplication)
            androidLogger()
        }

        // Warm up Nostr identity to ensure npub is available for favorite notifications
        try {
            com.bitchat.android.nostr.NostrIdentityBridge.getCurrentNostrIdentity(this)
        } catch (_: Exception) { }

        // Initialize theme preference
        ThemePreferenceManager.init(this)
    }
}
